esp/services/ws_dfu/ws_dfuService.cpp

```diff
@@ -1401,47 +1384,45 @@ bool CWsDfuEx::onDFUArrayAction(IEspContext &context, IEspDFUArrayActionRequest
             strncpy(curfile, file, len);
             curfile[len] = 0;
             
-            DBGLOG("CWsDfuEx::onDFUArrayAction User=%s Action=%s File=%s",username.str(),req.getType(), file);
             try
             {
-                onDFUAction(userdesc.get(), curfile, cluster, req.getType(), returnStr);
+                Owned<IDistributedFile> df = queryDistributedFileDirectory().lookup(curfile, userdesc.get(), true);
+                if (df)
+                {
+                    if (subfiles.length() > 0)
+                        subfiles.append(",");
+                    subfiles.append(curfile);
+                }
+                else
+                    errorStr.appendf("<Message><Value>%s not found</Value></Message>", curfile);
             }
             catch(IException* e)
             {
                 StringBuffer emsg;
                 e->errorMessage(emsg);
-                if((e->errorCode() == DFSERR_CreateAccessDenied) && (req.getType() != NULL))
-                {
-                    if (strcmp(req.getType(), Action_AddtoSuperfile) == 0)
-                    {
-                        emsg.replaceString("Create ", "AddtoSuperfile ");               
-                    }
-                }
+                if (e->errorCode() == DFSERR_CreateAccessDenied)
+                    emsg.replaceString("Create ", "AddtoSuperfile ");
 
-                returnStr.appendf("<Message><Value>%s</Value></Message>", emsg.str());
+                errorStr.appendf("<Message><Value>%s for %s</Value></Message>", emsg.str(), curfile);
                 e->Release();
             }
             catch(...)
             {
-                returnStr.appendf("<Message><Value>Unknown exception onDFUArrayAction %s</Value></Message>", curfile);
+                errorStr.appendf("<Message><Value>Unknown exception for %s</Value></Message>", curfile);
             }
             delete [] curfile;
         }
 
-        if (strcmp(Action_AddtoSuperfile ,req.getType()) == 0)
+        if (errorStr.length())
         {
-            returnStr.replaceString("#", "%23");
-            if (version < 1.18)
-                resp.setRedirectUrl(StringBuffer("/WsDFU/AddtoSuperfile?Subfiles=").append(returnStr.str()));
-            else
-            {
-                resp.setRedirectTo(returnStr.str());
-            }
```

Comment:
Would kind of rather see these changes in a separate pull request... unless it has something to do with the switch to enumeration?